modules/common/src/main/scala/surge/tracing/TracedMessage.scala

```diff
@@ -2,45 +2,18 @@
 
 package surge.tracing
 
-import java.util
-
 import com.fasterxml.jackson.annotation.JsonTypeInfo
-import io.opentracing.propagation.{ Format, TextMapExtract, TextMapInject }
-import io.opentracing.{ References, Span, SpanContext, Tracer }
+import io.opentracing.{ Span, Tracer }
 import surge.akka.cluster.JacksonSerializable
 
-import scala.jdk.CollectionConverters._
-
 object TracedMessage {
-  def apply[T](tracer: Tracer, message: T, span: Span): TracedMessage[T] = {
-    val headers: scala.collection.mutable.Map[String, String] = scala.collection.mutable.Map.empty[String, String]
-    tracer.inject(
-      span.context,
-      Format.Builtin.TEXT_MAP_INJECT,
-      new TextMapInject {
-        override def put(key: String, value: String): Unit = headers.put(key, value)
-      })
-    span.context()
-    TracedMessage(message, headers.toMap)
-  }
+
+  def apply[T](message: T, parentSpan: Span)(implicit tracer: Tracer): TracedMessage[T] =
+    TracedMessage(message, TracePropagation.asHeaders(parentSpan))
+
 }
 
-case class TracedMessage[T](
+final case class TracedMessage[T](
     @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "messageType", visible = true) message: T,
     headers: Map[String, String])
-    extends JacksonSerializable {
-  private def spanContext(tracer: Tracer): Option[SpanContext] = {
-    Option(
-      tracer.extract(
-        Format.Builtin.TEXT_MAP_EXTRACT,
```

Comment:
this wasn't working with Jaeger ;-( 
Jaeger only seems to like TEXT_MAP and not TEXT_MAP_EXTRACT AND TEXT_MAP_INJECT

Reply:
Interesting does it work with Zipkin as well? I'm good with whatever way it works as long as it works across the board :)

Reply:
Yes.. TEXT_MAP seems to work with both zipkin and jaeger ;-)